### DISCUSSION TITLE
[QUESTION] - How can I stop consumer from server?

### DISCUSSION THREAD

**Discussion by sujit-baniya** [nats-io/nats-server#4904] [Q&A] [answered]
Is it possible to stop consumer from consuming messages from the server?

**Comment by Jarema** [comment]
Consumers will consume messages only if clients are actively subscribed to them.
Currently there is no way to stop consumer in a way that it would halt processing even if there are active subscriptions/fetch requests for it if that is what you asked for.

**Reply by sujit-baniya** [reply]
Thanks for clarification. That was my question :). I was trying to implement Queue Worker with Nats where use could control the worker (consumer) to consume by Start/Stop on demand meanwhile Publisher could indefinitely enqueue tasks to the worker. Once the worker is resumed/started, it would start consuming the tasks again.

**Reply by jnmoyne** [reply, marked as answer]
Just use a pull consumer on your worker (using the old JS API), or use Messages() (new simplified JS API), it will only receive messages when it actively pulls them (by calling Fetch() or Messages())
